analyses/tmb-compare-tcga/compare-tmb.Rmd

```diff
@@ -60,186 +56,109 @@ if (!dir.exists(plots_dir)) {
 
 Custom function for plotting the TMB. 
 
+Function for making a combined CDF plot for TMB. 
+This plotting function was adapted from the [`breaks_cdf_plot` function in the 
+`chromosomal-instability`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/b1b73fe321a97fa82d85c86d20bd85635aabba25/analyses/chromosomal-instability/util/chr-break-plot.R#L120)
+
 ```{r}
-sina_plot <- function(data, colour = NULL, x_axis = "short_histology", title) {
-  # Given a data.frame with metadata and `tmb` information, make a sina plot from it
-  #
-  # Args:
-  #
-  # data : a data.frame with `tmb` as columns
-  # x_axis : A name of a column in data that you would like to make the x_axis from
-  #           Given as a character string. 
-  # colour: the color you would like the points to be
-  # title: The title for the plot. To be passed to `ggtitle`
-  #
-  # Returns:
-  #   A sina plot by histology group
-  data %>%
-    # eval parse bit will treat the variable as the text it has stored
-    ggplot2::ggplot(ggplot2::aes(x = reorder(eval(parse(text = x_axis)), tmb, mean), 
-                                 y = log10(tmb))) +
-    ggforce::geom_sina(color = colour, alpha = 0.3, maxwidth = 0.9) +
-    ggplot2::stat_summary(
-      fun.y = mean, fun.ymin = mean, fun.ymax = mean,
-      geom = "crossbar",
-      width = 0.95, size = 0.15
+tmb_cdf_plot <- function(tmb_df, plot_title, colour) {
+  tmb_df %>%
+    as.data.frame() %>%
+    dplyr::mutate(short_histology = tools::toTitleCase(short_histology)) %>%
+    # Only plot histologies groups with more than `min_samples` number of samples
+    dplyr::group_by(short_histology, add = TRUE) %>%
+    # Only keep groups with this amount of samples
+    dplyr::filter(dplyr::n() > 5) %>%
+    # Calculate histology group mean
+    dplyr::mutate(
+      hist_mean = mean(tmb),
```

Comment:
I think we should switch to median here for both the horizontal line and ordering. We will still easily be able to see outliers, but they will not affect ordering as much, which can make a big difference given the smaller sample size of the tcga data.

Reply:
Switched to median.